features/filesystem/fat/FATFileSystem.cpp

```diff
@@ -660,7 +660,24 @@ void FATFileSystem::dir_seek(fs_dir_t dir, off_t offset) {
     FATFS_DIR *dh = static_cast<FATFS_DIR*>(dir);
 
     lock();
-    dh->index = offset;
+    if (offset < dh->index) {
+        f_rewinddir(dh);
+    }
+    while (dh->index < offset) {
+        FILINFO finfo;
+        FRESULT res;
+#if _USE_LFN
+        char lfname[NAME_MAX];
+        finfo.lfname = lfname;
+        finfo.lfsize = NAME_MAX;
+#endif // _USE_LFN
+        res = f_readdir(dh, &finfo);
```

Comment:
Return value/error here will not be propagated to application.

Reply:
The only useful thing an application can do after dir_seek() is dir_read(), at which point it will receive an error.